pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1336,16 +1336,13 @@ func (ctrl *PersistentVolumeController) provisionClaimOperation(claimObj interfa
 	// Bind it to the claim
 	volume.Spec.ClaimRef = claimRef
 	volume.Status.Phase = v1.VolumeBound
+	if volume.Spec.StorageClassName == "" {
```

Comment:
The code here makes sure that provisioned PV has the class set correctly - either by the provisioner (`provisioner.Provision()` call on line 1321), or if the provisioner forgets to set it, the controller does it here.